In [14]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.ticker as tck
sns.set(style="darkgrid")

# ***Coleta e tratamento dos dados***

In [15]:
# Fazendo o upload dos conjuntos de dados:
df_dados = pd.read_csv('dataset/data_2000_2019.csv', encoding='utf-8')
df_metadados = pd.read_csv('dataset/data_desc.csv', encoding='ISO-8859-1')

In [16]:
# Verificando as 5 primeiras e últimas linhas dos conjunto de dados
display(df_dados)
display(df_metadados)

micro_code        micro_name   micro_name_ibge  meso_code  \
0            11001       Porto Velho       PORTO VELHO       1101   
1            11002  Guajar<e1>-Mirim     GUAJARÁ-MIRIM       1101   
2            11003         Ariquemes         ARIQUEMES       1102   
3            11004      Ji-Paran<e1>         JI-PARANÁ       1102   
4            11005  Alvorada D'Oeste  ALVORADA D'OESTE       1102   
...            ...               ...               ...        ...   
133915       52015        Meia Ponte        MEIA PONTE       5205   
133916       52016      Pires do Rio      PIRES DO RIO       5205   
133917       52017        Catal<e3>o           CATALÃO       5205   
133918       52018   Quirin<f3>polis      QUIRINÓPOLIS       5205   
133919       53001       Bras<ed>lia          BRASÍLIA       5301   

                meso_name  state_code        state_name  region_code  \
0         Madeira-Guaporé          11          Rondônia            1   
1         Madeira-Guaporé          11          Rondônia            1   
2       Leste Rondoniense          11          Rondônia            1   
3       Leste Rondoniense          11          Rondônia            1   
4       Leste Rondoniense          11          Rondônia            1   
...                   ...         ...               ...          ...   
133915         Sul Goiano          52             Goiás            5   
133916         Sul Goiano          52             Goiás            5   
133917         Sul Goiano          52             Goiás            5   
133918         Sul Goiano          52             Goiás            5   
133919   Distrito Federal          53  Distrito Federal            5   

        region_name  biome_code  ... time  dengue_cases population  \
0             North           5  ...    1           NaN        NaN   
1             North           5  ...    1           NaN        NaN   
2             North           5  ...    1           NaN        NaN   
3             North           5  ...    1           NaN        NaN   
4             North           5  ...    1           NaN        NaN   
...             ...         ...  ...  ...           ...        ...   
133915  Centre-West           2  ...  240         111.0   409691.0   
133916  Centre-West           2  ...  240          42.0   100780.0   
133917  Centre-West           2  ...  240          57.0   176527.0   
133918  Centre-West           2  ...  240          41.0   128463.0   
133919  Centre-West           2  ...  240         576.0  3015268.0   

       pop_density       tmax       tmin      pdsi  urban  water_network  \
0              NaN  30.544556  21.751202 -1.790602  85.20          33.57   
1              NaN  31.005884  21.641306  0.629408  72.31          27.43   
2              NaN  30.444357  21.492065 -3.455422  66.91          27.76   
3              NaN  30.458706  21.111828 -1.586286  68.68          32.23   
4              NaN  30.616793  21.053421 -0.043228  42.82          21.40   
...            ...        ...        ...       ...    ...            ...   
133915   19.397264  31.148733  21.699646 -2.526940  89.69          79.10   
133916   10.579541  29.929796  20.871635 -1.855514  72.80          71.34   
133917   11.591627  30.154041  20.992001 -2.663716  89.47          88.74   
133918    7.991280  32.152596  22.195051 -3.365225  85.36          77.39   
133919  521.289159  28.822147  19.971571 -1.842697  96.77          94.82   

        water_shortage  
0               0.2245  
1               0.0873  
2               0.1871  
3               0.2381  
4               0.2837  
...                ...  
133915          0.4656  
133916          0.2444  
133917          0.6508  
133918          0.4802  
133919          0.8730  

[133920 rows x 26 columns]

Variable                                       Long name  \
0        micro_code                                Microregion code   
1        micro_name                                Microregion name   
2   micro_name_ibge                 Microregion name (IBGE sources)   
3         meso_code                                 Mesoregion code   
4         meso_name                                 Mesoregion name   
5        state_code                                      State code   
6        state_name                                      State name   
7       region_code                                     Region code   
8       region_name                                     Region name   
9        biome_code                                      Biome code   
10       biome_name                                      Biome name   
11     ecozone_code                                    Ecozone code   
12     ecozone_name                                    Ecozone name   
13     main_climate                             Main climate regime   
14            month                                           Month   
15             year                                            Year   
16             time                                            Time   
17     dengue_cases                           Notified dengue cases   
18       population                                      Population   
19      pop_density                              Population density   
20             tmax                            Maximium temperature   
21             tmin                             Minimum temperature   
22             pdsi  Self-calbrated Palmer's drought severity index   
23            urban                                Urban population   
24    water_network                         Access to water network   
25   water_shortage                                  Water shortage   

                                          Description  \
0   Unique ID number given by IBGE to each microre...   
1                            Name of each microregion   
2        Name of each microregion (from IBGE sources)   
3   Unique ID number given by IBGE to each mesoreg...   
4                             Name of each mesoregion   
5        Unique ID number given by IBGE to each state   
6                                      Name of state    
7                                         Region code   
8                                         Region name   
9                                         Biome code    
10                                         Biome name   
11  Ecozone code (same as biome apart from Atlanti...   
12  Ecozone name (same as biome apart from Atlanti...   
13  Most prevalent climate regime in the microregi...   
14  Calendar month index, 1 = January, É, 12 = Dec...   
15                                   Year 2000 - 2019   
16          Time index starting at 1 for January 2000   
17  Number of notified dengue cases registered in ...   
18  Estimated population, based on projections cal...   
19      Population density (number of people per km2)   
20  Monthly average daily maximum temperature; gri...   
21  Monthly average daily minimum temperature; gri...   
22  Self-calibrated Palmer drought severity index ...   
23  Percentage of inhabitants living in urban area...   
24  Percentage of inhabitants with access to the p...   
25  Frequency of reported water shortages per micr...   

                                               Source  
0   Data from https://www.ibge.gov.br/geociencias/...  
1   Data from https://www.ibge.gov.br/geociencias/...  
2   Data from https://www.ibge.gov.br/geociencias/...  
3   Data from https://www.ibge.gov.br/geociencias/...  
4   Data from https://www.ibge.gov.br/geociencias/...  
5   Data from https://www.ibge.gov.br/geociencias/...  
6   Data from https://www.ibge.gov.br/geociencias/...  
7   Data from https://www.ibge.gov.br/geociencias/...  
8   Data from https://www.ibge.gov.br/geociencias

In [17]:
df_dados.columns


Index(['micro_code', 'micro_name', 'micro_name_ibge', 'meso_code', 'meso_name',
       'state_code', 'state_name', 'region_code', 'region_name', 'biome_code',
       'biome_name', 'ecozone_code', 'ecozone_name', 'main_climate', 'month',
       'year', 'time', 'dengue_cases', 'population', 'pop_density', 'tmax',
       'tmin', 'pdsi', 'urban', 'water_network', 'water_shortage'],
      dtype='object')

In [22]:
lista_variaveis = df_metadados.Variable.to_list()
lista_descricoes = df_metadados.Description.to_list()

print(lista_variaveis)
print(lista_descricoes)

['micro_code', 'micro_name', 'micro_name_ibge', 'meso_code', 'meso_name', 'state_code', 'state_name', 'region_code', 'region_name', 'biome_code', 'biome_name', 'ecozone_code', 'ecozone_name', 'main_climate', 'month', 'year', 'time', 'dengue_cases', 'population', 'pop_density', 'tmax', 'tmin', 'pdsi', 'urban', 'water_network', 'water_shortage']
['Unique ID number given by IBGE to each microregion (5 digits)', 'Name of each microregion', 'Name of each microregion (from IBGE sources)', 'Unique ID number given by IBGE to each mesoregion (4 digits)', 'Name of each mesoregion', 'Unique ID number given by IBGE to each state', 'Name of state ', 'Region code', 'Region name', 'Biome code ', 'Biome name', 'Ecozone code (same as biome apart from Atlantic Rainforest, biome code 6, which has been separated into NE Atlantic rainforest, ecozone code 6, SE Atlantic Rainforest, ecozone code 7, and S Alantic Rainforest, ecozone code 8)', 'Ecozone name (same as biome apart from Atlantic Rainforest, which 

In [24]:
valor = 0
# Verificando a quantidade de colunas no df_dados, listando o nome e a descricao de cada uma
print(f"Temos {len(df_dados.columns)} colunas no conjunto de dados, conforme a seguir:")
for colunas in df_dados.columns:
    if colunas == lista_variaveis[valor]:
        print(f"{colunas}: {lista_descricoes[valor]}")
    valor = valor+1

Temos 26 colunas no conjunto de dados, conforme a seguir:
*micro_code*: Unique ID number given by IBGE to each microregion (5 digits)
*micro_name*: Name of each microregion
*micro_name_ibge*: Name of each microregion (from IBGE sources)
*meso_code*: Unique ID number given by IBGE to each mesoregion (4 digits)
*meso_name*: Name of each mesoregion
*state_code*: Unique ID number given by IBGE to each state
*state_name*: Name of state 
*region_code*: Region code
*region_name*: Region name
*biome_code*: Biome code 
*biome_name*: Biome name
*ecozone_code*: Ecozone code (same as biome apart from Atlantic Rainforest, biome code 6, which has been separated into NE Atlantic rainforest, ecozone code 6, SE Atlantic Rainforest, ecozone code 7, and S Alantic Rainforest, ecozone code 8)
*ecozone_name*: Ecozone name (same as biome apart from Atlantic Rainforest, which has been separated into NE Atlantic rainforest, SE Atlantic Rainforest, and S Alantic Rainforest)
*main_climate*: Most prevalent climat

**Descrição do conteúdo das colunas:**    
- **micro_name_ibge**: Nome da microrregião, conforme definido pelo Instituto Brasileiro de Geografia e Estatística (IBGE).    
- **meso_name**: Nome da mesorregião, uma divisão geográfica que agrupa microrregiões com características sociais e econômicas semelhantes.    
- **state_name**: Nome do estado onde os dados foram coletados.    
- **region_name**: Nome da região geográfica do Brasil (Norte, Nordeste, Centro-Oeste, Sudeste, Sul) onde os dados foram coletados.    
- **biome_name**: Nome do bioma em que a coleta de dados ocorreu, como Amazônia, Cerrado, Caatinga, etc.    
- **ecozone_name**: Nome da zona ecológica, que pode se referir a uma classificação mais específica dentro de um bioma ou região.    
- **main_climate**: Classificação climática principal da região conforme categorizada.    
- **dengue_cases**: Número de casos de dengue registrados na região durante o período de coleta de dados.    
- **population**: População total na região durante o período de coleta de dados.    
- **pop_density**: Densidade populacional na região (habitantes por quilômetro quadrado).    
- **tmax**: Temperatura máxima registrada (possivelmente a média mensal ou anual) na região durante o período de coleta de dados.    
- **tmin**: Temperatura mínima registrada (possivelmente a média mensal ou anual) na região durante o período de coleta de dados.    
- **pdsi**: Índice de Severidade de Seca de Palmer (Palmer Drought Severity Index), que mede a secura ou umidade relativa do solo.    
- **urban**: Porcentagem ou classificação da área considerada urbana na região.    
- **water_network**: Indicador da presença ou qualidade da rede de abastecimento de água na região.    
- **water_shortage**: Indica se houve escassez de água na região durante o período de coleta de dados.

In [25]:
# Excluindo colunas que não serão utilizadas nas análises posteriores
df_dados_limpo = df_dados.drop(["micro_code","micro_name", "meso_code", "region_code", "biome_code", "ecozone_code","time", "population","pop_density"], axis=1)

# Verificando as 5 primeiras e últimas linhas do novo conjunto de dados
display(df_dados_limpo)

micro_name_ibge          meso_name  state_code        state_name  \
0            PORTO VELHO    Madeira-Guaporé          11          Rondônia   
1          GUAJARÁ-MIRIM    Madeira-Guaporé          11          Rondônia   
2              ARIQUEMES  Leste Rondoniense          11          Rondônia   
3              JI-PARANÁ  Leste Rondoniense          11          Rondônia   
4       ALVORADA D'OESTE  Leste Rondoniense          11          Rondônia   
...                  ...                ...         ...               ...   
133915        MEIA PONTE         Sul Goiano          52             Goiás   
133916      PIRES DO RIO         Sul Goiano          52             Goiás   
133917           CATALÃO         Sul Goiano          52             Goiás   
133918      QUIRINÓPOLIS         Sul Goiano          52             Goiás   
133919          BRASÍLIA   Distrito Federal          53  Distrito Federal   

        region_name biome_name ecozone_name main_climate  month  year  \
0             North     Amazon       Amazon           AM      1  2000   
1             North     Amazon       Amazon           AM      1  2000   
2             North     Amazon       Amazon           AM      1  2000   
3             North     Amazon       Amazon           AM      1  2000   
4             North     Amazon       Amazon           AM      1  2000   
...             ...        ...          ...          ...    ...   ...   
133915  Centre-West    Cerrado      Cerrado           AW     12  2019   
133916  Centre-West    Cerrado      Cerrado           AW     12  2019   
133917  Centre-West    Cerrado      Cerrado           AW     12  2019   
133918  Centre-West    Cerrado      Cerrado           AW     12  2019   
133919  Centre-West    Cerrado      Cerrado           AW     12  2019   

        dengue_cases       tmax       tmin      pdsi  urban  water_network  \
0                NaN  30.544556  21.751202 -1.790602  85.20          33.57   
1                NaN  31.005884  21.641306  0.629408  72.31          27.43   
2                NaN  30.444357  21.492065 -3.455422  66.91          27.76   
3                NaN  30.458706  21.111828 -1.586286  68.68          32.23   
4                NaN  30.616793  21.053421 -0.043228  42.82          21.40   
...              ...        ...        ...       ...    ...            ...   
133915         111.0  31.148733  21.699646 -2.526940  89.69          79.10   
133916          42.0  29.929796  20.871635 -1.855514  72.80          71.34   
133917          57.0  30.154041  20.992001 -2.663716  89.47          88.74   
133918          41.0  32.152596  22.195051 -3.365225  85.36          77.39   
133919         576.0  28.822147  19.971571 -1.842697  96.77          94.82   

        water_shortage  
0               0.2245  
1               0.0873  
2               0.1871  
3               0.2381  
4               0.2837  
...                ...  
133915          0.4656  
133916          0.2444  
133917          0.6508  
133918          0.4802  
133919          0.8730  

[133920 rows x 17 columns]

In [26]:
# Verificando o tamanho do conjunto de dados

df_dados_limpo.shape

(133920, 17)

In [27]:
# Verificando se existem duplicatas

duplicatas = df_dados_limpo.duplicated()
duplicatas.sum()

0

In [28]:
# Verificando se existem valores nulos e quais são os tipos de dados de cada coluna do conjunto

df_dados_limpo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 133920 entries, 0 to 133919
Data columns (total 17 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   micro_name_ibge  133920 non-null  object 
 1   meso_name        133920 non-null  object 
 2   state_code       133920 non-null  int64  
 3   state_name       133920 non-null  object 
 4   region_name      133920 non-null  object 
 5   biome_name       133920 non-null  object 
 6   ecozone_name     133920 non-null  object 
 7   main_climate     133920 non-null  object 
 8   month            133920 non-null  int64  
 9   year             133920 non-null  int64  
 10  dengue_cases     127224 non-null  float64
 11  tmax             133920 non-null  float64
 12  tmin             133920 non-null  float64
 13  pdsi             133680 non-null  float64
 14  urban            133920 non-null  float64
 15  water_network    133920 non-null  float64
 16  water_shortage   133920 non-null  floa

- Podemos notar que o conjunto de dados possui 133920 linhas, porem:
    - a coluna *dengue_cases* possui 127224 registros;
    - a coluna *pdsi* possui 133680 registros.

- Com isso, vamos verificar quais sao os valores faltantes.

In [30]:
missing_values = df_dados_limpo.isnull().sum()
print(missing_values)

micro_name_ibge       0
meso_name             0
state_code            0
state_name            0
region_name           0
biome_name            0
ecozone_name          0
main_climate          0
month                 0
year                  0
dengue_cases       6696
tmax                  0
tmin                  0
pdsi                240
urban                 0
water_network         0
water_shortage        0
dtype: int64


Conforme resultado, realmente possuimos valores nulos nas colunas *dengue_cases* e *pdsi*

- Para *dengue_cases*: vou considerar casos nulos como 0, assumindo que a ausência desse registro pode significar 0 casos reportados;
- Para *pdsi*: vou substituir os valores nulos com o valor médio da coluna.

In [32]:
# Tratamento da coluna 'dengue_cases'
df_dados_limpo['dengue_cases'] = df_dados_limpo['dengue_cases'].fillna(0)

# Tratamento da coluna 'pdsi'
media_pdsi = df_dados_limpo['pdsi'].mean()
df_dados_limpo['pdsi'] = df_dados_limpo['pdsi'].fillna(media_pdsi)

# Verificando novamente a presença de valores faltantes após o tramento
missing_values_fill = df_dados_limpo.isnull().sum()
print(missing_values_fill)

micro_name_ibge    0
meso_name          0
state_code         0
state_name         0
region_name        0
biome_name         0
ecozone_name       0
main_climate       0
month              0
year               0
dengue_cases       0
tmax               0
tmin               0
pdsi               0
urban              0
water_network      0
water_shortage     0
dtype: int64


In [33]:
# Verificando os dados estatísticos do conjunto de dados (registros, média, valor mínimo e máximo, etc)
df_dados_limpo.describe()

state_code          month           year   dengue_cases  \
count  133920.000000  133920.000000  133920.000000  133920.000000   
mean       31.302867       6.500000    2009.500000      96.291017   
std        10.385255       3.452065       5.766303     946.859279   
min        11.000000       1.000000    2000.000000       0.000000   
25%        24.000000       3.750000    2004.750000       0.000000   
50%        31.000000       6.500000    2009.500000       4.000000   
75%        41.000000       9.250000    2014.250000      26.000000   
max        53.000000      12.000000    2019.000000   88821.000000   

                tmax           tmin           pdsi          urban  \
count  133920.000000  133920.000000  133920.000000  133920.000000   
mean       29.417284      18.980845      -0.459147      72.098190   
std         3.461089       3.644395       1.905470      16.630274   
min        14.425291       4.615391      -7.634145      22.300000   
25%        27.555401      16.923784      -1.747564      58.390000   
50%        30.162950      19.892204      -0.562458      72.750000   
75%        31.774971      21.770352       0.714378      86.680000   
max        37.885544      25.765200       7.479940     100.000000   

       water_network  water_shortage  
count  133920.000000   133920.000000  
mean       72.286989        0.357296  
std        17.021309        0.240645  
min        16.680000        0.000000  
25%        62.680000        0.160100  
50%        74.875000        0.325850  
75%        85.700000        0.531700  
max        98.810000        1.000000

# ***Análise Exploratória dos dados***